lambdas/service/app.py

```diff
@@ -222,44 +222,163 @@ def openapi_raw():
+
 
 @app.route('/health', methods=['GET'], cors=True, method_spec={
-    **health_spec,
-    'summary': 'List health information for webservice'
+    'summary': 'Complete health check',
+    'description': format_description('''
+        Health check of the service and all resources it depends on. This may
+        take long time to complete and exerts considerable load on the service.
+        For that reason it should not be requested frequently or by automated
+        monitoring facilities that would be better served by the
+        [`/health/fast`](#operations-Auxiliary-get_health_fast) endpoint.
+    '''),
+    **health_spec(health_all_keys)
 })
+def health():
+    return health_controller().full_response()
+
+
+@app.route('/health/basic', methods=['GET'], cors=True, method_spec={
+    'summary': 'Basic health check',
+    'description': format_description('''
+        Health check of only the REST API itself, excluding other resources
+        the service depends on. A 200 response indicates that the service is
+        reachable via HTTP(S) but nothing more.
+    '''),
+    **health_spec(health_up_key)
+})
+def basic_health():
+    return health_controller().basic_response()
+
+
+@app.route('/health/cached', methods=['GET'], cors=True, method_spec={
+    'summary': 'Cached health check for continuous monitoring',
+    'description': format_description('''
+        Return a cached copy of the
+        [`/health/fast`](#operations-Auxiliary-get_health_fast) response.
+        This endpoint is optimized for continuously running distributed health
+        monitoring facilities such as AWS CloudWatch. The caching ensures that
+        this service is not overloaded by those facilities. The cache is updated
+        every minute.
+    '''),
+    **health_spec(health_service_keys)
+})
+def cached_health():
+    return health_controller().cached_response()
+
+
+@app.route('/health/fast', methods=['GET'], cors=True, method_spec={
+    'summary': 'Fast health check',
+    'description': format_description('''
+        Performance-optimized health check of the REST API and critical
+        resources the service depends on. This endpoint can be requested more
+        frequently than [`/health`](#operations-Auxiliary-get_health) but
+        periodic and automated requests should be made to
+        [`/health/fast`](#operations-Auxiliary-get_health_fast).
+    '''),
+    **health_spec(health_service_keys)
+})
+def fast_health():
+    return health_controller().fast_response()
+
+
 @app.route('/health/{keys}', methods=['GET'], cors=True, path_spec={
     'parameters': [
         {
             'name': 'keys',
             'in': 'path',
             'required': True,
-            'schema': {'type': 'string'}
+            'description': f'Available keys to query health status: {[*health_controller().all_keys]}',
+            'schema': schema.array(str,
+                                   pattern=f'^({"|".join(health_controller().all_keys)})'
```

Comment:
I saw no evidence of the pattern anywhere in the Swagger UI so it should be ripped out.